import "libnutrition/proto/data/composite_food.proto";
import "libnutrition/proto/service/error.proto";

message CompositeFoodLoadRequest 
{
	optional bool omit = 1 [default = false];
	optional bool omitExpired = 2 [default = false];
	optional bool nameAndIdOnly = 3 [default = false];
	optional bool includeDatesInName = 4 [default = false];
	repeated int32 requestedIds = 5;
	repeated int32 requestedUserIds = 6;
}

message CompositeFoodLoadResponse
{
	repeated CompositeFoodData compositeFoods = 1;
	optional Error error = 100;
}

message CompositeFoodStoreRequest
{
	repeated CompositeFoodData compositeFoods = 1;
}

message CompositeFoodStoreResponse
{
	repeated int32 storedIds = 1;
	optional Error error = 100;
}

message CompositeFoodDeleteRequest
{
	repeated int32 deleteIds = 1;
}

message CompositeFoodDeleteResponse
{
	repeated int32 deletedIds = 1;
	optional Error error = 100;
}

